packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts

```diff
@@ -29,6 +29,59 @@ export type TestOperation = (
 	random: IRandom,
 ) => IMergeTreeOp | undefined;
 
+export const insertField: TestOperation = (
+	client: TestClient,
+	opStart: number,
+	opEnd: number,
+	random: IRandom,
+) => {
+	const numberText = random.string(5, "0123456789");
+	if (
+		// start is not a number
+		Number.isNaN(Number(client.getText(opStart, opStart + 1))) &&
+		// previous is not a number
+		(opStart === 0 ||
+			opStart >= client.getLength() ||
+			Number.isNaN(Number(client.getText(opStart - 1, opStart))))
+	) {
+		return client.insertTextLocal(opStart, numberText);
+	}
+};
+
+const numberRange = (
```

Comment:
i would probably call this something like `getFieldExtents` and return `{start, end} | undefined` and use this everywhere you inspect to see if a position is a field. if it is, you get the start and end, if it is not, you get undefined.